File: spending-api/models/store.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Store struct {
	Id         int
	UUId       uuid.UUID
	Name       string
	CategoryId int
	CreatedAt  time.Time
	UpdatedAt  time.Time
	IsDeleted  bool
	DeletedAt  time.Time
}

func NewStore(name string, categoryId int) *Store {
	return &Store{
		UUId:       uuid.New(),
		Name:       name,
		CategoryId: categoryId,
		CreatedAt:  time.Now().UTC(),
		UpdatedAt:  time.Now().UTC(),
	}
}

File: spending-ui/src/models/store.ts
export interface StoreDto {
    Id: string;
    Name: string;
    CreatedAt: string;
    UpdatedAt: string;
}

export interface Store {
    id: string;
    name: string;
    createdAt: Date;
    updatedAt: Date;
}

export function mapStoreFromDto(dto: StoreDto): Store {
    return {
        id: dto.Id,
        name: dto.Name,
        createdAt: new Date(dto.CreatedAt),
        updatedAt: new Date(dto.UpdatedAt),
    };
}

export interface CreateStoreDto {
    name: string;
}

export interface UpdateStoreDto {
    id: string;
    name: string;
}